userspace/plugin/plugin_types.h

```diff
@@ -47,6 +48,11 @@ typedef enum ss_plugin_field_type {
 	// The field encodes only the IP address, so this differs from FTYPE_IPADDR,
 	// from the way the framework perform runtime checks and comparisons.
 	FTYPE_IPNET = 41,
+	// A Go/Python-style zero-indexed start and end of the field's
+	// location in the event data as a uint32_t pair. {0,0} can be used
+	// by extractors to indicate that the field doesn't correspond to a
+	// location.
+	FTYPE_METADATA_SLICE = 1000000,
```

Comment:
How exactly is this encoded? Extracted values go via the anonymous union in ss_plugin_extract_field and I can't see a new variant there. Can't say I like the current layout of the type (I'd much prefer a ss_plugin_state_data* instead of union of pointers) but we should have the format documented as a C struct nonetheless.

Reply:
My PoC used res.u32.